ansible_collections/arista/avd/roles/eos_designs/templates/core_interfaces/logic.j2

```diff
@@ -122,6 +122,10 @@
 {%                     do core_interfaces_data.bgp_neighbors.update({ p2p_interface.peer_ip: p2p_bgp_neighbor }) %}
 {%                 endif %}
 {%             elif switch.underlay_routing_protocol in ["ospf", "ospf-ldp"] %}
+{%                 set p2p_interface.ospf_area = p2p_link.ospf_area | arista.avd.default(link_info.p2p_profile.ospf_area, underlay_ospf_area) %}
+{%                 set p2p_interface.ospf_authentication = p2p_link.ospf_authentication_mode | arista.avd.default(link_info.p2p_profile.ospf_authentication_mode, none) %}
```

Comment:
I think the main problem of everything not being rendered is here. you are using `ospf_authentication_mode` variables but according to the README, the expectation for the key is `ospf_authentication` (simple | message-digest) and so it is not going to work. 

Additionally the documentation states that the default is "simple" but here it would be done. Maybe there is no default because by default authentication is off ? (and in that case let's remove the default statement from documentation)

The following change should make the rest of the PR work:

```suggestion
{%                 set p2p_interface.ospf_authentication = p2p_link.ospf_authentication | arista.avd.default(link_info.p2p_profile.ospf_authentication, none) %}
```

But we may want to align with the `isis` variable name which is `isis_authentication_mode` and in that case use `ospf_authentication_mode` everywhere.